pkg/vm/engine/tae/txn/txnimpl/localseg.go

```diff
@@ -101,10 +101,12 @@ func (seg *localSegment) ApplyAppend() (err error) {
 		if destOff, err = ctx.driver.ApplyAppend(
 			bat,
 			seg.table.store.txn); err != nil {
+			ctx.driver.UnRef()
```

Comment:
1. Don't do UnRef
2. ApplyAppend may not be called and the data block will never be released.